src/Expr.ml

```diff
@@ -66,6 +68,28 @@ type synthesized = {
   outputs : Value.t array ;
 } [@@deriving sexp]
 
+let unify_component (comp : component) (args : synthesized list) : component option =
+  let typs = List.map args ~f:(fun s -> Value.typeof s.outputs.(0)) in
+  (* Log.debug (lazy ( "Unifying (" ^ (List.to_string_map ~sep:"," ~f:Type.to_string comp.domain)
+                    ^ ") of (" ^ comp.name ^ ") with ("
+                    ^ (List.to_string_map ~sep:"," ~f:(fun a -> (Type.to_string a)) typs)
+                    ^ ")")); *)
```

Comment:
You could remove these commented lines, or (improve them if needed and) uncomment them -- logging shouldn't hurt performance since its disabled by default.

Reply:
I dont think i need it anymore, I have removed it. If needed I will add it again later.